src/test/rpc/Feature_test.cpp

```diff
@@ -0,0 +1,274 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <test/jtx.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/app/misc/AmendmentTable.h>
+
+namespace ripple {
+
+class Feature_test : public beast::unit_test::suite
+{
+    void
+    testNoParams()
+    {
+        testcase ("No Params, None Enabled");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature") [jss::result];
+        if(! BEAST_EXPECT(jrr.isMember(jss::features)))
+            return;
+        for(auto const& feature : jrr[jss::features])
+        {
+            //default config - so all should be disabled, not vetoed, and supported
```

Comment:
The existence of `jss::name` in the result should probably be checked at least once in the unit test.
`if(! BEAST_EXPECT(feature.isMember(jss::name)))
    return;`